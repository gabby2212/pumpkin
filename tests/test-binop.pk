print(3 + 4)
print(3 - 4)
print(3 * 4)
print(3 / 4)
print(3 % 4)
print(3 == 4)
print(3 != 4)
print(3 > 4)
print(3 < 4)
print(3 <= 4)
print(3 >= 4)

print(3.5 + 4.2)
print(3.5 - 4.5)
print(3.5 * 4.5)
print(3.5 / 4.0)
print(3.5 % 4.5)
print(3.5 == 4.5)
print(3.5 != 4.5)
print(3.5 > 4.5)
print(3.5 < 4.5)
print(3.5 <= 4.5)
print(3.5 >= 4.5)

print("test1" + "test2")
print("test1" * "test2")
print("test1" == "test2")
print("test1" != "test2")
print("test1" > "test2")
print("test1" < "test2")
print("test1" <= "test2")
print("test1" >= "test2")

print('t' + 'b')
print('t' * 'b')
print('t' == 'b')
print('t' != 'b')
print('t' > 'b')
print('t' < 'b')
print('t' <= 'b')
print('t' >= 'b')

print(True && False)
print(True || False)
print(True == False)
print(False != False)

print(2::[3, 4])