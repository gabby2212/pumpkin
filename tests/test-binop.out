7
-1
12
0.75
3
20
20
75
-3
false
true
false
true
true
false
7.7
-1
15.75
0.875
3.5
false
true
false
true
true
false
test1test2
NaN
false
true
false
true
true
false
tb
NaN
false
true
true
false
false
true
false
true
false
false
[ 2, 3, 4 ]
